cmd/tcl/kubectl-testkube/devbox/devutils/objectstorage.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//	https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package devutils
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	minio2 "github.com/minio/minio-go/v7"
+
+	"github.com/kubeshop/testkube/cmd/testworkflow-toolkit/artifacts"
+	"github.com/kubeshop/testkube/internal/common"
+	"github.com/kubeshop/testkube/pkg/log"
+	"github.com/kubeshop/testkube/pkg/storage/minio"
+)
+
+type ObjectStorage struct {
+	pod            *PodObject
+	localPort      int
+	hashes         map[string]string
+	hashMu         sync.RWMutex
+	cachedClient   *minio2.Client
+	cachedClientMu sync.Mutex
+}
+
+func NewObjectStorage(pod *PodObject) *ObjectStorage {
+	return &ObjectStorage{
+		pod:    pod,
+		hashes: make(map[string]string),
+	}
+}
+
+func (r *ObjectStorage) Is(path string, hash string) bool {
+	r.hashMu.RLock()
+	defer r.hashMu.RUnlock()
+	return r.hashes[path] == hash
+}
+
+func (r *ObjectStorage) SetHash(path string, hash string) {
+	r.hashMu.Lock()
+	defer r.hashMu.Unlock()
+	r.hashes[path] = hash
+}
+
+func (r *ObjectStorage) Create(ctx context.Context) error {
+	err := r.pod.Create(ctx, &corev1.Pod{
+		Spec: corev1.PodSpec{
+			TerminationGracePeriodSeconds: common.Ptr(int64(1)),
+			Containers: []corev1.Container{
+				{
+					Name:            "minio",
+					Image:           "minio/minio:RELEASE.2024-10-13T13-34-11Z",
```

Comment:
there are a lot of diffferent hardcodings, const values in the code like this. might be it's ok for dev tool, but definitely not for prod code